### ISSUE TITLE
zql: put can't copy unset values

### ISSUE THREAD

**Issue by henridf** [brimsec/zq#846]
Given `put newfield=somefield`, for a record that has `somefield` but with unset value, then I would expect that the output record would have an unset `newfield`. Yet:

```sh
$ cat unsetin.tzng 
#1:record[col:string]
1:[a;]
1:[-;]
$ zq -t "put b = col" unsetin.tzng 
col: value is unset
#0:record[col:string,b:string]
0:[a;a;]
0:[-;]
```

In the second output record, no `-` was added for b. 

(As I pasted this I realized there's also a more immediate bug: this output is invalid tzng as the second record is missing a field. Not filing a separate bug as these seem tightly related).

**Comment by henridf** [comment]
This is an expression of the broader #850, but I filed it as a separate one as has a smaller scope and may be somewhat more pressing.

**Comment by philrz** [comment]
I can verify that this was fixed in `zq` commit `56ed80d` that's associated with #1375. The output now includes the unset value for the field.

```
$ zq -t "put b = col" unsetin.tzng
#0:record[col:string,b:string]
0:[a;a;]
0:[-;-;]
```

Thanks @mccanne!
